src/App.tsx

```diff
@@ -1,61 +1,111 @@
+import React, { useState } from 'react';
 import './App.scss';
+import users from './api/users';
+import todos from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './interfaces/todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const newId = (list: Todo[]) => {
+  const maxId = Math.max(...list.map(todo => todo.id));
+
+  return maxId + 1;
+};
+
+export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [currentUserId, setCurrentUserId] = useState('0');
+  const [invalidTitle, setInvalidTitle] = useState(false);
+  const [invalidUser, setInvalidUser] = useState(false);
+
+  const onTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setInvalidTitle(false);
+  };
+
+  const onUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setCurrentUserId(event.target.value);
+    setInvalidUser(false);
+  };
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setInvalidTitle(true);
+    }
+
+    if (currentUserId === '0') {
+      setInvalidUser(true);
+    }
+
+    if (!title.trim() || currentUserId === '0') {
+      return;
+    }
```

Comment:
Better add return in each if
```suggestion
    if (!title.trim()) {
      setInvalidTitle(true);
            return;
    }

    if (currentUserId === '0') {
      setInvalidUser(true);
            return;
    }
```

Reply:
Add check into change method. Any case if you have questions - ask them in fe_chat